File: LHDS.AdminPortal.Web/ClientApp/src/models/audits/audit.ts
import { Guid } from 'guid-typescript';

export class Audit {
    public id: Guid;
    public ingestionTrackingId: string;
    public message: string;
    public createdDate: Date;
    public createdBy: string;
    public updatedDate: Date;
    public updatedBy: string;

    constructor(audit: any) {
        this.id = audit.id ? Guid.parse(audit.id) : Guid.parse(Guid.EMPTY);
        this.ingestionTrackingId = audit.ingestionTrackingId;
        this.message = audit.message || "";
        this.createdDate = audit.createdDate ? new Date(audit.createdDate) : new Date();
        this.createdBy = audit.createdBy || "";
        this.updatedDate = audit.updatedDate ? new Date(audit.updatedDate) : new Date();
        this.updatedBy = audit.updatedBy || "";
    }
}

File: LHDS.AdminPortal.Api.Tests.Acceptance/Models/Audits/Audit.cs
// ---------------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------------

using System;
using System.Text.Json.Serialization;
using Newtonsoft.Json.Converters;

namespace LHDS.AdminPortal.Api.Tests.Acceptance.Models.Audits
{
    public class Audit
    {
        public Guid Id { get; set; }
        public Guid IngestionTrackingId { get; set; }
        public string Message { get; set; }
        public string CreatedBy { get; set; }
        public string UpdatedBy { get; set; }

        [JsonConverter(typeof(IsoDateTimeConverter))]
        public DateTimeOffset UpdatedDate { get; set; }


        [JsonConverter(typeof(IsoDateTimeConverter))]
        public DateTimeOffset CreatedDate { get; set; }
    }
}
